src/main/java/randoop/output/FailingTestFilter.java

```diff
@@ -0,0 +1,235 @@
+ * to be run on regression tests to filter tests that pass within Randoop, but fail when run from
+ * the command line.
+ *
+ * <p>Writes the class, and then compiles and runs the tests to determine whether there are failing
+ * assertions. Each failing assertion is replaced by a comment containing the code for the failing
+ * assertion. Creates a clean temporary directory for each compilation/run of a test class to avoid
+ * state effects due to files in the working directory.
+ */
+public class FailingTestFilter implements CodeWriter {
+
+  /** A pattern matching the JUnit4 message indicating the total count of failures */
+  private static final Pattern FAILURE_MESSAGE_PATTERN =
+      Pattern.compile("There\\s+(?:was|were)\\s+(\\d+)\\s+failure(?:s|):");
+
+  /** Regex for Java identifiers */
+  private static final String ID_STRING =
+      "\\p{javaJavaIdentifierStart}\\p{javaJavaIdentifierPart}*";
+
+  /** The {@link randoop.execution.TestEnvironment} for running the test classes. */
+  private final TestEnvironment testEnvironment;
+
+  /** The underlying {@link randoop.output.JavaFileWriter} for writing a test class. */
+  private final JavaFileWriter javaFileWriter;
+
+  /**
+   * Create a {@link FailingTestFilter} run in the environment and using the given {@link
+   * JavaFileWriter} to output test classes.
+   *
+   * @param testEnvironment the {@link TestEnvironment} for executing tests during filtering
+   * @param javaFileWriter the {@link JavaFileWriter} to write java files for the classes
+   */
+  public FailingTestFilter(TestEnvironment testEnvironment, JavaFileWriter javaFileWriter) {
+    this.testEnvironment = testEnvironment;
+    this.javaFileWriter = javaFileWriter;
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * <p>Inserts comments in place of assertions that fail when the test is run.
+   *
+   * <p>May be sensitive to changes in JUnit runner output.
+   */
+  @Override
+  public File writeClass(String packageName, String classname, String classString) {
+
+    List<File> sourceList = new ArrayList<>();
+    File testFile = javaFileWriter.writeClass(packageName, classname, classString);
+    sourceList.add(testFile);
+
+    String qualifiedClassname = (packageName.isEmpty() ? "" : packageName + ".") + classname;
+
+    int pass = 0;
+    boolean passing = false;
+
+    while (!passing) {
+
+      Path workingDirectory = createWorkingDirectory(classname, pass);
+
+      boolean success = compileTestClass(sourceList, workingDirectory);
+      assert success;
+
+      RunStatus status = null;
+      try {
+        status = testEnvironment.runTest(qualifiedClassname, workingDirectory.toFile());
+      } catch (ProcessException e) {
+        throw new BugInRandoopException("Error filtering regression tests: " + e.getMessage());
+      }
+      if (status.exitStatus == 0) {
+        passing = true;
+        continue;
+      }
+
+      /*
```

Comment:
This method body is long, with no overview of what is happening.  I suggest that you abstract out the code from here to line 173 into a separate method.  With a useful name and documentation, this will make both it and the code that calls it more comprehensible.